core/include/crypto/crypto_se.h

```diff
@@ -15,6 +15,28 @@
 
 #include <tee_api_types.h>
 
+/*
+ * Type identifier for the APDU message as described by Smart Card Standard
+ * ISO7816-4 about ADPU message bodies decoding convention:
+ *
+ * https://cardwerk.com/smart-card-standard-iso7816-4-section-5-basic-organizations/#chap5_3_2
+ */
+enum crypto_apdu_type {
+	CRYPTO_APDU_CASE_NO_HINT,
+	CRYPTO_APDU_CASE_1,
+	CRYPTO_APDU_CASE_2,
+	CRYPTO_APDU_CASE_2E,
+	CRYPTO_APDU_CASE_3,
+	CRYPTO_APDU_CASE_3E,
+	CRYPTO_APDU_CASE_4,
+	CRYPTO_APDU_CASE_4E,
+};
+
+TEE_Result crypto_se_do_apdu(enum crypto_apdu_type type,
```

Comment:
This function declared here, but only defined in the next commit. I believe this could lead to a link error while bisecting.

Reply:
um, well ok, will push the se050 driver first...but this should not be enabled without a driver anyway

Reply:
what I mean is this: https://github.com/OP-TEE/optee_os/pull/4935/commits/72df9ed590e76c589ba72325be87319596c54fa9 ...now (after I swapped the commits) this commit will be enabling a config that does nothing

Reply:
so not sure what is worse TBH...probably what I just pushed is worse but let me know what you prefer

Reply:
The new order is fine, but I think that the changes in this file belongs in the commit "crypto: se050: APDU driver".

Reply:
you are asking me to remove crypto_se.h from this commit?

Reply:
No, I'm expecting that you provide a declaration for `crypto_se_do_apdu()` when you introduce the global definition `crypto_se_do_apdu()` in the same commit.

Reply:
That is totally unnecessary. I'll revert it to the way it was: building the PTA only makes sense when the definition is provided

Reply:
Noted

Reply:
or if you let me know how to to generate the bisect error I'll fix of course! I just dont want to add unnecessary code...OP-TEE is complex as it is....